### DISCUSSION TITLE
Watch-Mode ignores test filter

### DISCUSSION THREAD

**Discussion by SvenAndBits** [vitest-dev/vitest#6954] [Q&A] [answered]
Hi,

I’m experiencing some unexpected behavior with Vitest when files are modified. It’s possible this is intentional or due to a recent change, but I wanted to clarify the expected behavior.

**Scenario:**

1. Vitest is started with the command: `vitest watch feature-a`.
2. Vitest executes the tests in `feature-a.spec.ts` (assuming only one match for `*feature-a*` exists in this case).
3. A change is made in another test file, `feature-b.spec.ts`.

**Expected Behavior:**
Vitest should only re-execute the tests in `feature-a.spec.ts`, since the initial command explicitly targeted that file.

**Actual Behavior:**
Vitest re-executes the tests in `feature-a.spec.ts`, but it also runs tests in other files, such as `feature-b.spec.ts`.

Additionally, I’ve observed similar behavior when non-test code is modified. If the modified code is covered by other test suites, those suites are also executed unexpectedly.

Is this the intended behavior, or could there be an issue with my setup? Any guidance would be appreciated.

Best regards

**Comment by hi-ogawa** [comment, marked as answer]
I think that's the behavior as long as I know, but I'm not sure if there's any specific reason :thinking:
I made a PR to change it and will see what happens https://github.com/vitest-dev/vitest/pull/6955

**Reply by SvenAndBits** [reply]
Hi @hi-ogawa,

Thank you for your reply!

I truly appreciate your support and the effort you put into creating a PR. Based on the initial response, it does seem like this might be a regression.

I’m hopeful that your PR will be accepted and merged soon—thanks again for addressing this!
